### PR TITLE
Fix find on-chain checkpoint

### PR DESCRIPTION
**Pull request by claravanstaden**
There were 2 bugs in the find on-chain checkpoint method:
- The check constrains the checkpoint to be in the same sync committee period, which is not a requirement (a checkpoint can span over 2 sync committee periods, for example, header 8288 can prove slots 8192 - 8288 in period 1 and slots 96 - 1892 in period 0).
- The search does not traverse the ring buffer correctly (it only searches from the latest index to 0. It should wrap around and search to index + 1)

Added new config in Ansible scripts: https://github.com/Snowfork/snowbridge-infra-suite/pull/24/files#diff-0bdba71d49616abfe1a14b43ead54c6df64cdbfca68c8e4e5fe47bd4a1e3ae2bR29

### CODE REVIEW

File: relayer/relays/beacon/config/config.go [hunk 1]
```diff
@@ -35,8 +34,14 @@ type BeaconConfig struct {
...
-	Parachain          config.ParachainConfig `mapstructure:"parachain"`
-	UpdateSlotInterval uint64                 `mapstructure:"updateSlotInterval"`
+	Parachain          ParachainConfig `mapstructure:"parachain"`
+	UpdateSlotInterval uint64          `mapstructure:"updateSlotInterval"`
+}
+
+type ParachainConfig struct {
+	Endpoint             string `mapstructure:"endpoint"`
+	MaxWatchedExtrinsics int64  `mapstructure:"maxWatchedExtrinsics"`
+	HeaderRedundancy     uint64 `mapstructure:"headerRedundancy"`
```
Comment (claravanstaden) [inline relayer/relays/beacon/config/config.go:44]: Setting to capture the max headers in the ringbuffer: // https://github.com/paritytech/polkadot-sdk/blob/master/bridges/snowbridge/pallets/ethereum-client/src/lib.rs#L75

File: relayer/relays/beacon/header/header.go [hunk 2]
```diff
@@ -523,32 +523,32 @@ func (h *Header) findLatestCheckPoint(slot uint64) (state.FinalizedHeader, error
...
 	}
 	startIndex := uint64(lastIndex)
-	endIndex := uint64(0)
+	endIndex := startIndex + 1
 
 	syncCommitteePeriod := h.protocol.Settings.SlotsInEpoch * h.protocol.Settings.EpochsPerSyncCommitteePeriod
-	slotPeriodIndex := slot / syncCommitteePeriod
-
-	for index := startIndex; index >= endIndex; index-- {
+	totalStates := syncCommitteePeriod * h.protocol.HeaderRedundancy // Total size of the circular buffer,
```
Comment (yrong) [inline relayer/relays/beacon/header/header.go:529]: Is total size of the circular buffer be 256*20=5120?

```
totalStates := h.protocol.Settings.EpochsPerSyncCommitteePeriod * h.protocol.HeaderRedundancy
```

File: web/packages/test/config/beacon-relay.json [hunk 3]
```diff
@@ -18,7 +18,8 @@
   "sink": {
     "parachain": {
       "endpoint": "ws://127.0.0.1:11144",
-      "maxWatchedExtrinsics": 8
+      "maxWatchedExtrinsics": 8,
+      "headerRedundancy": 20
```
Comment (yrong) [inline web/packages/test/config/beacon-relay.json:22]: Can we just move the new config `headerRedundancy` to `config.SpecSettings`, then not much change required.

Comment (claravanstaden) [reply web/packages/test/config/beacon-relay.json:22]: That's true, but it is not really related to the Ethereum spec settings, which is why I didn't add it there.

File: relayer/relays/beacon/config/config.go (hunk 1 above)
Comment (vgeddes) [reply relayer/relays/beacon/config/config.go:44]: Can we add a comment in the code describing the config item?

**Review by vgeddes — APPROVED** [review]
Looks good just plz add a comment describing the new config item.

File: relayer/relays/beacon/config/config.go (hunk 1 above)
Comment (claravanstaden) [reply relayer/relays/beacon/config/config.go]: Sure, https://github.com/Snowfork/snowbridge/pull/1294/commits/1956d878b677cd2a3b74b1d7313c27b3c15e6e87.

**Pull request merged**